Sentry.Xamarin.Forms/SentryXamarinFormsIntegration.cs

```diff
@@ -45,41 +50,53 @@ public void Register(IHub hub, SentryOptions options)
                 }
             });
 
-            if (LogXamlErrors)
+            if (Options.Value.XamarinLoggerEnabled)
             {
-                Log.Listeners.Add(XamlLogger);
+                Log.Listeners.Add(XamarinLogger);
             }
 
-            //If initialized from the Android/IOS project, the current application is not going to be set in time, so wait a bit...
-            Task.Run(async () =>
+            if (Options.Value.NativeIntegrationEnabled)
             {
-                for (int i = 0; i < 5 && Application.Current is null; i++)
-                {
-                    await Task.Delay(1000);
-                }
-                if (Application.Current is null)
+                if (new NativeIntegration(Options.Value) is NativeIntegration nativeIntegration &&
```

Comment:
I don't follow, so an instance is created and then you check if it's of that type?
It'll always be true.

What does `Implemented` here do? This code is hard to reason about

Reply:
Implemented says if the NativeIntegration was implemented (it was not implemented on NetStandard 2.0) while the IOS and Android returns true